### PR TITLE
adapt policy to APIM 3.15+

### PR DESCRIPTION
**Pull request by phiz71**
**Issue**

gravitee-io/issues#8385

**Description**

bump gravitee-parent, gravitee-common, gravitee-gateway-api and adapt code.

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `2.0.0-adapt-3-15-x-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/policy/gravitee-policy-traffic-shadowing/2.0.0-adapt-3-15-x-SNAPSHOT/gravitee-policy-traffic-shadowing-2.0.0-adapt-3-15-x-SNAPSHOT.zip)

### CODE REVIEW

File: src/main/java/io/gravitee/policy/trafficshadowing/TrafficShadowingPolicy.java [hunk 1]
```diff
@@ -57,65 +59,94 @@ public ReadWriteStream<Buffer> onRequestContent(ExecutionContext context) {
...
+                .forEach(header -> {
+                    if (header.getName() != null && !header.getName().trim().isEmpty()) {
+                        try {
+                            String extValue = (header.getValue() != null) ? context.getTemplateEngine().convert(header.getValue()) : null;
+                            if (extValue != null) {
+                                httpHeaders.set(header.getName(), extValue);
                             }
+                        } catch (Exception ex) {
+                            // Do nothing
+                            ex.printStackTrace();
```
Comment (jhaeyaert) [inline src/main/java/io/gravitee/policy/trafficshadowing/TrafficShadowingPolicy.java:86]: ⚠️

File: src/main/java/io/gravitee/policy/trafficshadowing/TrafficShadowingPolicy.java [hunk 2]
```diff
@@ -57,65 +59,94 @@ public ReadWriteStream<Buffer> onRequestContent(ExecutionContext context) {
...
+                            response.bodyHandler(buffer -> {}).endHandler(handler -> {});
+                        });
+                    }
+                );
+        }
+
+        @Override
+        public SimpleReadWriteStream<Buffer> write(Buffer chunk) {
+            if (connection == null) {
+                chunkBuffers.add(chunk);
```
Comment (jhaeyaert) [inline src/main/java/io/gravitee/policy/trafficshadowing/TrafficShadowingPolicy.java:122]: We should calculate the total retained size here (ie sum buffer.length) and if it exceed a certain size, just give up to preserve memory (maybe log a warn)

**Review by jhaeyaert — CHANGES_REQUESTED** [review]
I've made some propositions to improve the resilience.
If implemented, they should be documented.

**Comment by a-cordier** [comment]
see https://github.com/gravitee-io/gravitee-policy-traffic-shadowing/pull/21

**Pull request closed without merging**